core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2591,14 +2598,18 @@ private void loadMetadataAndJournal(RpcContext rpcContext, LockedInodePath inode
             }
             try (TempInodePathForChild tempInodePath =
                 new TempInodePathForChild(inodePath, childStatus.getName())) {
-              DescendantType loadDescendantType =
-                  (options.getLoadDescendantType() == DescendantType.ONE) ? DescendantType.NONE :
-                      DescendantType.ALL;
               LoadMetadataOptions loadMetadataOptions =
-                  LoadMetadataOptions.defaults().setLoadDescendantType(loadDescendantType)
-                      .setCreateAncestors(false).setUfsStatus(childStatus);
+                  LoadMetadataOptions.defaults().setLoadDescendantType(DescendantType.NONE)
+                      .setCreateAncestors(true).setUfsStatus(childStatus);
```

Comment:
What happens if a UFS path `/a/b/c/foo` gets "loaded" before `/a/b/`? The first path would have already created `/a/b/` (create ancestor is true), but not with a UFS status. What if the actual Ufs status for `/a/b/` is different?

Reply:
I was trying to get away with not sorting and relying on creating ancestor to do the work for me. but i think it is actually kind of complicated. Instead I will sort the Ufs paths so no children is visited before their parent.